Geometry/CaloTopology/src/CaloTowerConstituentsMap.cc

```diff
@@ -94,35 +96,64 @@ std::vector<DetId> CaloTowerConstituentsMap::constituentsOf(const CaloTowerDetId
   if (standardHB_) {
     if (id.ietaAbs()<=m_cttopo->lastHBRing()) {
       m_hcaltopo->depthBinInformation(HcalBarrel,hcal_ieta,id.iphi(),id.zside(),nd,sd);
-      for (int i=0; i<nd; i++)
-        items.push_back(HcalDetId(HcalBarrel,hcal_ieta*id.zside(),id.iphi(),i+sd));
+      for (int i=0; i<nd; i++) {
+	HcalDetId hid = m_hcaltopo->mergedDepthDetId(HcalDetId(HcalBarrel,hcal_ieta*id.zside(),id.iphi(),i+sd));
+        if (std::find(items.begin(),items.end(),hid) == items.end()) {
+	  items.push_back(hid);
```

Comment:
actually, is this at all relevant to plan-1, considering that the checked id is in HcalBarrel?
Same question for the HO case below.